### PR TITLE
chore: migrate fixtures for core_tests.py

### PR DESCRIPTION
**Pull request by hughhhh**
### SUMMARY

Cleaning up the test objects in core_tests, and moving them into a separate file for organization. Hoping this can help with isolation moving forward.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: tests/core_tests.py [hunk 1]
```diff
@@ -508,13 +513,7 @@ def test_testconn_failed_conn(self, username="admin"):
...
         )
 
-        data = json.dumps(
-            {
-                "uri": "mssql+pymssql://url",
-                "name": "examples",
-                "impersonate_user": False,
-            }
-        )
+        data = bad_database_test_conn_request()
```
Comment (willbarrett) [inline tests/core_tests.py:516]: This could become a true Pytest fixture

File: tests/core_tests.py [hunk 2]
```diff
@@ -892,68 +891,7 @@ def test_explore_json(self):
...
-            "columns": [],
-            "row_limit": 10,
-            "color_scheme": "supersetColors",
-            "label_colors": {},
-            "show_legend": True,
-            "y_axis_format": "SMART_NUMBER",
-            "bottom_margin": "auto",
-            "x_ticks_layout": "auto",
-        }
+        form_data = dist_bar_form_data(table_id=tbl_id)
```
Comment (willbarrett) [inline tests/core_tests.py:894]: This could be moved to a Pytest fixture that performs the load on the previous line

**Review by willbarrett — COMMENTED** [review]
Let's see how much of this we can move to true Pytest fixtures rather than loading the data via function calls in the test implementation.

File: tests/fixtures/core.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                "schema": None,
+                "autorun": False,
+                "sql": "SELECT ...",
+                "queryLimit": 1000,
+            }
+        )
+    }
+
+
+def dist_bar_form_data(table_id):
```
Comment (betodealmeida) [inline tests/fixtures/core.py:41]: In pytest fixtures don't have an easy way of accepting parameters. You can either modify the response from the fixture like I suggested for the bad connection, or you can [parametrize it](https://docs.pytest.org/en/latest/example/parametrize.html#indirect-parametrization):

```python
@pytest.fixture
def dist_bar_form_data(request):
    table_id = request.param
    return { .... }  # note the use of return here
```

Then:

```python
@pytest.mark.parametrize('dist_bar_form_data', [[tbl_id]], indirect=True)
def test_create_query_editor(dist_bar_form_data):
    ...
```

Maybe there's a better way?

**Review by betodealmeida — CHANGES_REQUESTED** [review]
Hugh, IMHO some of the fixtures here (eg, `bad_database_test_conn_request`) are too specific to be reused, they would benefit from being more generic. I also left some notes on how to make them into pytest fixtures, so they can be reused more easily.

**Pull request closed without merging**